src/main/kotlin/com/ampnet/identityservice/controller/VeriffController.kt

```diff
@@ -0,0 +1,100 @@
+package com.ampnet.identityservice.controller
+
+import com.ampnet.identityservice.controller.pojo.VeriffRequest
+import com.ampnet.identityservice.exception.VeriffException
+import com.ampnet.identityservice.service.VeriffService
+import com.ampnet.identityservice.service.VerificationService
+import com.ampnet.identityservice.service.pojo.ServiceVerificationResponse
+import mu.KLogging
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestHeader
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder
+import javax.servlet.http.HttpServletRequest
+
+@RestController
+class VeriffController(
+    private val veriffService: VeriffService,
+    private val verificationService: VerificationService
+) {
+
+    companion object : KLogging()
+
+    @Suppress("ReturnCount")
+    @PostMapping("/veriff/session")
+    fun getVeriffSession(
+        @RequestBody request: VeriffRequest,
+        servlet: HttpServletRequest
+    ): ResponseEntity<ServiceVerificationResponse> {
+        val address = ControllerUtils.getAddressFromSecurityContext()
+        val baseUrl = ServletUriComponentsBuilder.fromRequestUri(servlet)
+            .replacePath(null)
+            .build()
+            .toUriString()
+        logger.info { "Received request to get veriff session for address: $address" }
+        val payloadValid = verificationService.verifyPayload(address, request.signedPayload)
+        if (payloadValid.not()) return ResponseEntity.badRequest().build()
```

Comment:
In this flow, JWT is used to verify user identity so there is no need to verify signed payload.

Reply:
https://linear.app/ampnet/issue/SD-136/store-kyc-data-from-veriff, how come? Why are passing the signed_payload then in the request?

Reply:
This was the first specification for the flow without JWT. If we use JWT to verify user requests then we don't need to double-check it with payload signature.

Reply:
Ok